### ISSUE TITLE
curation status message on publication pages

### ISSUE THREAD

**Issue by ValWood** [pombase/website#858]
Re conversation with JAcky:

>Val I started curating PMID:10725227 and Saz was looking it up in pombase and it says it has been curated by pombase staff.   No problem, Ill leave it for now I just wanted to let you know and I wondered if it was completed.

me:
it's one that I did back in the day before we had Canto and phenotype ontologies. Ideally these would not say
"curated by PomBase staff" but  something like " This paper has only legacy curation, if you would like to add more detail  please
get in touch via the help-desk" Otherwise it just looks like we made a really bad job!

discuss updating text.... no hurry

**Comment by ValWood** [comment]
Should we change the ones out for CC from


This publication has no annotations in PomBase because it hasn't been curated yet. (Please contact us if you would like to contribute annotations for this paper.)


" This paper has been sent out for community curation" or similar.  To indicate "ball in their court"

**Comment by ValWood** [comment]
@mar @Antonialock  thoughts?

**Comment by mah11** [comment]
I don't have a strong preference here ...

**Comment by Antonialock** [comment]
Yeah I think it wouldn't hurt.

Maybe 'this paper hasn't yet been curated for all data types. If you would like to..."
and
" This paper has been sent out for community curation, if you would like to..."
(obviously we wouldn't reassign something really new but if a paper has been out for years and nothing has happened...)

**Comment by ValWood** [comment]
Actually, I think these 2 messages are sufficient.
OK?

**Comment by mah11** [comment]
Well, I think the first of Val's suggestions in https://github.com/pombase/website/issues/858#issuecomment-426722837 is a bit wordy, and "helpdesk" shouldn't be hyphenated because we write it as "helpdesk" everywhere else.

But I also think 3 messages would be overkill.

How about:

- "This paper has partial curation that predates PomBase and Canto. Please contact the curators if you would like to contribute to its full curation" (with the usual mailto link for "contact us")

- "Community curation is in progress for this paper. Please contact the curators if you would like to suggest annotations from it"

**Comment by ValWood** [comment]
perfect!

**Comment by ValWood** [comment]
Almost quick text change

**Comment by ValWood** [comment]
@kimrutherford  this looks quick?

**Comment by ValWood** [comment]
It would be good to do this because if a paper is curated by BIOGRID for interactions is appears as
Curated by PomBase staff

even when it is not curated. misleading  because we didn't make the annotations, but might also make people think we have done it and it does not need curating

**Comment by kimrutherford** [comment]
> This paper has only legacy curation

Is that when there is annotation by PomBase but there is no Canto data?

I'm quite confused by this discussion.  Are these the cases we need to cover?:

 - has community curation (ie. there's an approved community Canto session)
 - has PomBase curation, but only legacy - "This paper has partial curation that predates PomBase and Canto" (there's PomBase annotation, but no approved Canto session)
 - has PomBase curation, not legacy (there's an approved non-community Canto session)
 - has other annotation, like BIOGRID (there's only non-PomBase annotation, no Canto sessions)
 - "Community curation is in progress for this paper" (there's a Community curation session but it's not approved yet)
 - has no annotation

**Comment by ValWood** [comment]
I think the 2 messages  suggested by Midori are sufficient:

"This paper has partial curation that predates PomBase and Canto. Please contact the curators if you would like to contribute to its full curation" (with the usual mailto link for "contact us")

is also fine for
 - has PomBase curation, not legacy (there's an approved non-community Canto session)
(we don't mind if people contatc us while we are curating an old paper, but they never have)
- has other annotation, like BIOGRID (there's only non-PomBase annotation, no Canto sessions)

no message required as there is a full publication page
 - has community curation (ie. there's an approved community Canto session)
- has PomBase curation, not legacy (there's an approved non-community Canto session)

This message is only for papers with status "curatable".  Other publication types already have a suitable message

ie
This publication has no annotations in PomBase because it is a review.

**Comment by kimrutherford** [comment]
> This message is only for papers with status "curatable".

Sorry for being dumb but I still don't understand.  Which message do you mean?  Midori suggested two messages.

**Comment by ValWood** [comment]
Corrected to

"These messages are only for papers with status curatable"

Does that make more sense?

**Comment by ValWood** [comment]
We can discuss this tomorrow ;)

**Comment by ValWood** [comment]
If curation status = curatable

1. Triaged, curatable
No message
-  has community curation (ie. there's an approved community Canto session)
- has PomBase curation, not legacy (there's an approved non-community Canto session)

2.Legacy curation (some curation but no Canto session)
- has PomBase curation, but only legacy -  (there's PomBase annotation, but no approved Canto session)
- has other annotation, like BIOGRID (there's only non-PomBase annotation, no Canto sessions)

Message
"This paper has partial curation that predates PomBase and Canto. Please contact the curators if you would like to contribute to its full curation" (with the usual mailto link for "contact us")

3. Uncurated, but paper sent out to community curator (there's a Community curation session but it's not approved yet)

Message
"Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it"

**Comment by mah11** [comment]
Logic implications:

- if triage status = review: no change
- if triage status = method: no change
- otherwise,
  - if triage status = curatable: go through the loop in the previous comment
  - otherwise, no change

**Comment by kimrutherford** [comment]
For this case:

> "Community curation is in progress for this paper, Please contact the curators
> if you would like to suggest annotations from it"

does there need to be a different message if curation is in progress by an admin?  Should we change "Community curation is in progress ..." to "Curation is in progress ..." in that case?

**Comment by kimrutherford** [comment]
Also I think there's a missing case that doesn't fit any of 1-4 above:

  - Legacy annotation and a (community or admin) session in progress.

Should that have the same message as 3?

**Comment by kimrutherford** [comment]
> No annotation, but paper sent out to community curator

For this case do you mean?:
No annotation, but community session exists and isn't approved yet

There is a least one like that that hasn't been sent out:
https://www.pombase.org/reference/PMID:9042863
https://curation.pombase.org/pombe/view/object/pub/6652?model=track

**Comment by kimrutherford** [comment]
Do JBrowse tracks count as legacy curation here?:
  - Legacy curation (some curation but no Canto session)
  - Legacy curation and a (community or admin) session in progress.

Here are the publications we have tracks for that don't have accepted sessions.  Some have annotations as well.  Do any of these need a message?  Currently none do.  Only the first two have sessions.

```
     pmid      | canto_curator_name |    canto_triage_status    | canto_annotation_status 
---------------+--------------------+---------------------------+-------------------------
 PMID:21981922 | François Bachand   | Curatable                 | SESSION_CREATED
 PMID:25665008 | Daniel Jeffares    | Curatable                 | CURATION_IN_PROGRESS
 PMID:22531001 |                    | Curatable                 | 
 PMID:24152550 |                    | Curatable                 | 
 PMID:31077324 |                    | Curatable                 | 
 PMID:30566651 |                    | Curatable                 | 
 PMID:24709818 |                    | Browser datasets, to host | 
(8 rows)
```

**Comment by kimrutherford** [comment]
Documentation for myself so I don't break the current behaviour:
 - there are Curatable publications with no annotation and an approved session.

In those cases we have this message:
This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types.

**Comment by kimrutherford** [comment]
As far as I understand, this is the logic needed to cover all the cases:

if Curatable:
  - if has annotations:
    - if has approved session:
        "Annotations from this publication:"
    - has a non-approved session:
       "Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."
    - has no session
       "This paper has partial curation that predates PomBase and Canto.  Please contact the curators if you would like to contribute to its full curation."
 - if no annotations:
   - has a approved canto session:
     "This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types."
   - has a non-approved session:
     "Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."
   - has no session:
     "This publication has no annotations in {{siteName}} because it hasn't been curated yet.  Please contact the curators if you want to contribute annotations for this paper."

if not Curatable:
  - Review or comment
    "This publication has no annotations in PomBase because it is a review."
  - Method or reagent
    "This publication has no annotations in PomBase because it is a methods paper.
  - all other cases:
    "This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types."

**Comment by kimrutherford** [comment]
> As far as I understand, this is the logic needed to cover all the cases:

I've edited the comment with example publications

**Comment by ValWood** [comment]
Wes,

For this one:

if curatable

>has a non-approved session:
(https://www.pombase.org/reference/PMID:15607976)
"Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."

>but maybe this needs a different message if admin is curating rather than community

If assigned to community use this message.
if assigned to Admin, it is likely we haven't started it so please use
"This paper has partial curation that predates PomBase and Canto. Please contact the curators if you would like to contribute to its full curation"


and again
has a non-approved session:
(https://www.pombase.org/reference/PMID:1699136)
"Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."

> but maybe this needs a different message if admin is curating rather than community

For admin assigned, no curation please use

"This publication has no annotations in PomBase because it hasn't been curated yet. Please contact the curators if you want to contribute annotations for this paper."

**Comment by ValWood** [comment]
>Do JBrowse tracks count as legacy curation here?:

I opened a curation tracker ticket for this.

**Comment by ValWood** [comment]
> There is a least one like that that hasn't been sent out:
https://www.pombase.org/reference/PMID:9042863
https://curation.pombase.org/pombe/view/object/pub/6652?model=track


This is an edge case and should disappear soon  ->APPROVED

**Comment by kimrutherford** [comment]
OK, thanks.  I've edited my summary.

**Comment by mah11** [comment]
I'll just plunk comments inline into your logic summary:

> As far as I understand, this is the logic needed to cover all the cases:
>
> if Curatable:
>   - if has annotations:
>     - if has approved session:
>         (https://www.pombase.org/reference/PMID:25908789)
>         "Annotations from this publication:"

yup

>     - has a non-approved session:
>       (https://www.pombase.org/reference/PMID:15607976)
>        - if it's a community session:
>           "Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."

Fix punctuation - don't use a comma between the sentences.

>        - if it's an admin session:
>           "This paper has partial curation that predates PomBase and Canto. Please contact the curators if you would like to contribute to its full curation"

I don't object, but another option is to leave out the word "community" from the start so the same message would do no matter who is doing the session. I.e. the message can be

"Curation is in progress for this paper. Please contact the curators if you would like to suggest annotations from it."

I'm not that bothered about whether we've got any further than creating a session for the purpose of pub page messages.

>     - has no session
>       (https://www.pombase.org/reference/PMID:16688222)
>        "This paper has partial curation that predates PomBase and Canto.  Please contact the curators if you would like to contribute to its full curation."

yup

>  - if no annotations:
>    - has an approved canto session:
>     (https://www.pombase.org/reference/PMID:8809106)
>      "This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types."

yup

>    - has a non-approved session:
>     (https://www.pombase.org/reference/PMID:1699136)
>      - if it's a community session:
>         "Community curation is in progress for this paper, Please contact the curators if you would like to suggest annotations from it."
>      - if it's an admin session:
>         "This publication has no annotations in PomBase because it hasn't been curated yet. Please contact the curators if you want to contribute annotations for this paper."

Same as above - fix punctuation; have the option to delete "community" so one message will do.

>    - has no session:
>     (https://www.pombase.org/reference/PMID:2630562)
>      "This publication has no annotations in PomBase because it hasn't been curated yet.  Please contact the curators if you want to contribute annotations for this paper."

yup

> if not Curatable:
>   - Review or comment
>     (https://www.pombase.org/reference/PMID:1680238)
>     "This publication has no annotations in PomBase because it is a review."
>   - Method or reagent
>     (https://www.pombase.org/reference/PMID:1107988)
>     "This publication has no annotations in PomBase because it is a methods paper.
>   - all other cases:
>     (https://www.pombase.org/reference/PMID:11032848)
>     "This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types."

yes to all "not curatable"

**Comment by mah11** [comment]
a couple more responses to some of Kim's other questions/comments:

>   - Legacy curation and a (community or admin) session in progress.
>
> Should that have the same message as 3?

We could either do that, or use a combined message, e.g.

"This paper has partial curation, and additional curation is in progress. Please contact the curators if you would like to suggest annotations from it."

I don't have a strong preference.


> Do JBrowse tracks count as legacy curation here?

Val's curation ticket is https://github.com/pombase/curation/issues/2610

Maybe we should have specific messages, e.g.

   - has active session
   "This paper is cited for browser tracks, and additional curation is in progress. Please contact the curators if you would like to suggest annotations from it."

   - no session
   "This paper is cited for browser tracks, but has not otherwise been curated. Please contact the curators if you would like to suggest annotations from it."

Or - a more radical suggestion: could pubs with browser tracks do without any other message, no matter what the status in Canto?

**Comment by ValWood** [comment]
I would like to mention on the ones that are community assigned but uncurated that they are "community curation in progress".  Many of these have not had sessions sent out for ages. This is because, if I sent out reminders for everything we would have too much to do.  So over the years, people might forget that they had an older session to curate. If they search for their own publication it is good if the message says "Community curation is in progress for this paper,..."  because it might prompt them to do it (otherwise they might think we are doing it)

**Comment by ValWood** [comment]
> Or - a more radical suggestion: could pubs with browser tracks do without any other message, no matter what the status in Canto?

I agree these don't require a message.
Although we need to check that all the browser hosted publications are otherwise fully curated, that's a curation issue

**Comment by kimrutherford** [comment]
I think I've finished implementing this.  It needs a nightly load to get the information from Chado that the website code needs.  I'll check after Friday night's load.

**Comment by ValWood** [comment]
I noticed that this paper
PMID:29891649
has no page but is in canto (SESSION_CREATED
(not sure what stage you are at though!)

**Comment by kimrutherford** [comment]
> I noticed that this paper
> PMID:29891649
> has no page but is in canto (SESSION_CREATED

It has triage status "Structure" so we don't export it from Canto for loading into Chado.

Here's the current list of triage statuses that get loaded into Chado:


https://github.com/pombase/canto/blob/d0826bbfa1393c3b71b83a6168ef721c9dd42f49/canto.yaml#L1300

**Comment by kimrutherford** [comment]
Here's the current list of triage statuses that get loaded into Chado:
  - Curatable
  - Browser datasets, to host
  - Browser datasets, hosted
  - Method or reagent
  - Review or comment
  - Mutagenicity or toxicity study
  - Biotech
  - Modelling
  - Phylogeny and evolutionary studies
  - Cell composition or WT feature
  - Not physically mapped
  - Other

**Comment by kimrutherford** [comment]
> I think I've finished implementing this. It needs a nightly load to get the information from Chado
> that the website code needs. I'll check after Friday night's load.

It's looking OK.

I'll go through the list on Monday on check the links: https://github.com/pombase/website/issues/858#issuecomment-586160647

**Comment by ValWood** [comment]
We should also include:

Modelling (81)
Biotech (37)
Database (30)
3D genome organization (9)
with
This publication has no annotations in PomBase because the data it contains either cannot be associated with individual genes, or cannot be represented by currently available annotation types.


and

Transposon related (28)
Mating-type related (19)
DNA replication related (27)
DNA recombination related (32)
Mitochondrial sequence related (14)

whatever the message is for "browser datasets to host"

**Comment by kimrutherford** [comment]
> Modelling (81)
> Biotech (37)

Those two are already in the configuration so should be displayed already.

> Database (30)
> 3D genome organization (9)

I've added those to the configuration for tonight's load.

> Transposon related (28)
> Mating-type related (19)
> DNA replication related (27)
> DNA recombination related (32)
> Mitochondrial sequence related (14)
> 
> whatever the message is for "browser datasets to host"

Currently for "Browser dataset, to host", if there are JBrowse tracks we show no message.  If there are no tracks in JBrowse (yet) we have the message:

> This publication has no annotations in PomBase because the data it contains either cannot be
> associated with individual genes, or cannot be represented by currently available annotation types.

That's from "all other cases" above (https://github.com/pombase/website/issues/858#issuecomment-586160647).

I've just added those triage types you list ("Transposon related", etc.) to the Canto export config so they'll show up in the main site tomorrow morning with the same "This publication has no annotations ..." message.

Should we have a have a separate message for publications that could one day have browser tracks?

**Comment by ValWood** [comment]
I suppose we could. We could also have a link to the instructions when we have them. Make it as easy as possible...

**Comment by kimrutherford** [comment]
> I suppose we could. We could also have a link to the instructions when we have them. Make it
> as easy as possible...

We could just add "or high throughput datasets" to the message if that would help, like this:

This publication has no annotations in PomBase because it hasn't been curated yet. Please contact the curators if you want to contribute annotations or high throughput datasets for this paper.

That change is ready to commit if it's useful.

**Comment by ValWood** [comment]
Yes do that and close. We can tweak later,

**Comment by kimrutherford** [comment]
> Yes do that and close. We can tweak later,

OK, done for tomorrow.

**Comment by kimrutherford** [comment]
> OK, done for tomorrow.

It's live now.  eg. https://www.pombase.org/reference/PMID:17053780
